README.md
# MSLaunchView
### 一键合成APP引导页，包含不同状态下的引导页操作方式，同时支持动态图片引导页和静态图片引导页,支持按钮跳过按钮，立即体验按钮完全自定义,PageControl样式多样（支持方形点，圆形点，横线+小圆点样式等),省掉冗余的代码,集成性高,使用方便;
---
#### 声明:部分图片来源于网络,如有涉及版权会马上删除,敬请谅解; 

### 效果图展示：
#### APP静态图片引导页(上) | APP动态图片引导页(下)
<table>
<tr>
<th>手动进入体验</th>
<th>自动进入体验</th>
</tr>
<tr>
<td><img src="https://github.com/lztbwlkj/MSLaunchView/blob/master/Demo/MSLaunchView/DesignSketchGIF/Untitled-1.gif" width="300"></td>
<td><img src="https://github.com/lztbwlkj/MSLaunchView/blob/master/Demo/MSLaunchView/DesignSketchGIF/Untitled-2.gif" width="300"></td>
</tr>
<tr>
<th>手动进入体验</th>
<th>自动进入体验</th>
</tr>
<tr>
<td><img src="https://github.com/lztbwlkj/MSLaunchView/blob/master/Demo/MSLaunchView/DesignSketchGIF/Untitled-6.gif" width="300"></td>
<td><img src="https://github.com/lztbwlkj/MSLaunchView/blob/master/Demo/MSLaunchView/DesignSketchGIF/Untitled-7.gif" width="300"></td>
</tr>    
</table>

#### [更新记录:](https://github.com/lztbwlkj/MSLaunchView/blob/master/CHANGELOG.md),

#### 使用方式（支持代码创建和StoryBoard创建项目）:

##### 图片数组（gif自动识别）:

###### 使用代码创建的项目,无进入立即按钮:
```objc
MSLaunchView *launchView = [MSLaunchView launchWithImages:imageNameArray isScrollOut:YES];
```

###### 使用StoryBoard创建的项目,无进入立即按钮:
```objc
MSLaunchView *launchView = [MSLaunchView launchWithImages:imageNameArray sbName:@"" isScrollOut:NO];
```

###### 使用代码创建的项目,有进入立即按钮:
```objc
MSLaunchView *launchView = [MSLaunchView launchWithImages:imageNameArray guideFrame:rt gImage:[UIImage imageNamed:@""] isScrollOut:NO]；
```


###### 使用StoryBoard创建的项目,有进入立即按钮:
```objc
MSLaunchView *launchView = [MSLaunchView launchWithImages:imageNameArray sbName:@"" guideFrame:rt gImage:[UIImage imageNamed:@""] isScrollOut:YES]；
```

##### 小视频（目前仅支持单个视频文件）:
```objc
NSString *path  = [[NSBundle mainBundle]  pathForResource:@"测试" ofType:@"mp4"];
NSURL *url = [NSURL fileURLWithPath:path];
[MSLaunchView launchWithVideo:CGRectMake(0, 0, MSScreenW, MSScreenH) videoURL:url];
```

==！！！如果你使用带立即进入按钮的初始化方法，那么后面的guideBtnCustom:方法将会失效！！！==

##### 视频的用法和图片的用法基本一致 这里就不再赘述,具体请参考Demo

### 集成方式:
#### 1.下载项目或者下载项目中MSLaunchView文件夹,将下载好的文件拖拽到自己的工程文件夹中,并在AppDelegate.h中导入#import "MSLaunchView.h"头文件;

#### 2. cocoapods集成:
###### 在项目的podfile文件中加入下面代码
```objc
pod 'MSLaunchView', '~>0.0.5'
```

#### 你还可以进行简单设置:
```objc
launchView.guideTitle = @"";
launchView.guideTitleColor = [UIColor redColor];

//是否隐藏跳过按钮
launchView.isHiddenSkipBtn = YES;
//PageControl属性
launchView.nomalColor = [UIColor lightGrayColor];
launchView.currentColor = [UIColor orangeColor];

//视频拉伸方式
launchView.videoGravity = AVLayerVideoGravityResize;
//播放完成是否自动推出 默认:YES
launchView.isPalyEndOut = NO;

```

#### 自定义按钮:
###### 自定义立即进入按钮:
```objc
[launchView guideBtnCustom:^UIButton * _Nonnull{
UIButton *btn = [UIButton buttonWithType:UIButtonTypeCustom];
btn.frame = CGRectMake(60, 60, 120, 120);
[btn setBackgroundColor:[UIColor redColor]];
[btn addTarget:self action:@selector(hidde) forControlEvents:UIControlEventTouchUpInside];
return btn;
}];

-(void)hidde{
[_launchView hideGuidView];
```

###### 自定义跳过按钮:
```objc
[launchView skipBtnCustom:^UIButton * _Nonnull{
UIButton *btn = [UIButton buttonWithType:UIButtonTypeCustom];
btn.frame = CGRectMake(60, 200, 120, 120);
[btn setBackgroundColor:[UIColor blueColor]];
[btn addTarget:self action:@selector(hidde) forControlEvents:UIControlEventTouchUpInside];
return btn;
}]；

-(void)hidde{
[_launchView hideGuidView];
}
```


## <<分享是一种美德,你的Star是我更新的动力，使用过程如果有什么问题可以[issues](https://github.com/lztbwlkj/MSLaunchView/issues/new),我会及时回复大家！>>




CHANGELOG.md
# 更新记录
###  2019.07.09 -- v0.0.9 
1、添加PageControl更多定义属性；
2、PageControl支持Pod自动管理；
3、新增最后一页是否隐藏PageControl属性(lastDotsIsHidden)；

### 2019.06.22 -- v0.0.8 
1、添加更多“跳过”按钮属性；
2、添加App是否是首次打开的宏定义；
3、微调初始化方法，是否测滑消失你说了算；
4、增加引导页隐藏后的回调方法； 

### 2019.01.05 -- v0.0.7:提交0.0.7版本 
1、添加PageControl更多的自定义属性;


Demo/MSLaunchView/AppDelegate.m
//
//  AppDelegate.m
//  MSLaunchView
//
//  Created by TuBo on 2018/11/8.
//  Copyright © 2018 TuBur. All rights reserved.
//

#import "AppDelegate.h"
#import "MSLaunchView.h"
#define MSScreenW   [UIScreen mainScreen].bounds.size.width
#define MSScreenH   [UIScreen mainScreen].bounds.size.height
@interface AppDelegate ()<MSLaunchViewDeleagte>{
    MSLaunchView *_launchView;
}

@end


@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
    
    self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
    self.window.backgroundColor = [UIColor whiteColor];
    UIViewController *vc = [[UIViewController alloc] init];
    vc.view.backgroundColor = [UIColor redColor];
    self.window.rootViewController = vc;
    [self.window makeKeyAndVisible];
    
    if (kisFirstLaunch) {//或者使用[MSLaunchView isFirstLaunch]判断是否是第一次进入
        
        NSArray *imageNameArray = @[@"Untitled-4.gif",@"Untitled-6.gif",@"Untitled-7.gif"];
        CGRect rt = CGRectMake(MSScreenW*0.3, MSScreenH*0.8, MSScreenW*0.4, MSScreenH*0.08);
        
#pragma mark ---->>带立即体验按钮，项目是SB创建的 (此方法初始化，则guideBtnCustom：方法失效)
//        MSLaunchView *launchView = [MSLaunchView launchWithImages:imageNameArray sbName:@"" guideFrame:rt gImage:[UIImage imageNamed:@""] isScrollOut:YES];
#pragma mark ---->>不带立即体验按钮 项目是SB创建的
//        MSLaunchView *launchView = [MSLaunchView launchWithImages:imageNameArray sbName:@"" isScrollOut:NO];
#pragma mark ---->>不带立即按钮 代码创建项目
//        MSLaunchView *launchView = [MSLaunchView launchWithImages:imageNameArray isScrollOut:YES];
#pragma mark ---->>带立即按钮 代码创建项目
        MSLaunchView *launchView = [MSLaunchView launchWithImages:imageNameArray guideFrame:rt gImage:[UIImage imageNamed:@""] isScrollOut:NO];

        
#pragma mark ---->>关于Video 没有立即进入按钮
//        NSString *path  = [[NSBundle mainBundle]  pathForResource:@"测试" ofType:@"mp4"];
//        NSURL *url = [NSURL fileURLWithPath:path];
//        MSLaunchView *launchView = [MSLaunchView launchWithVideo:CGRectMake(0, 0, MSScreenW, MSScreenH) videoURL:url];
//        launchView.videoGravity = AVLayerVideoGravityResize;
//        launchView.isPalyEndOut = YES;//
        
        launchView.guideTitle = @"进入当前界面";
        launchView.guideTitleColor = [UIColor blackColor];
        launchView.guideTitleFont = [UIFont systemFontOfSize:17];
        
#pragma mark ---->>跳过按钮自定义属性
        launchView.skipTitle = @"跳过";
        launchView.skipTitleColor = [UIColor whiteColor];
        launchView.skipTitleFont = [UIFont systemFontOfSize:15];
        launchView.skipBackgroundColor = [UIColor redColor];
        launchView.skipBackgroundImage = [UIImage imageNamed:@""];
        
        
#pragma mark ---->>PageControl自定义属性

//        launchView.showPageControl = NO;
        //pageControl的间距大小
//        launchView.pageControlStyle = MSPageControlStyleNumber;
     
//        launchView.pageControlBottomOffset += 25;
        launchView.pageDotColor = [UIColor redColor];
        launchView.currentPageDotColor = [UIColor yellowColor];
        launchView.textFont = [UIFont systemFontOfSize:9 weight:UIFontWeightBold];
        launchView.textColor = [UIColor blackColor];
        launchView.dotsIsSquare = NO;
        launchView.spacingBetweenDots = 15;
      
        launchView.dotBorderWidth = 2;
        launchView.dotBorderColor = [UIColor blueColor];
        launchView.currentDotBorderWidth = 2;
        launchView.currentDotBorderColor = [UIColor redColor];
        launchView.delegate = self;
        //设置了pageDotImage和currentPageDotImage图片 pageControlDotSize和currentWidthMultiple将失效
        launchView.pageControlDotSize = CGSizeMake(20, 20);
        launchView.currentWidthMultiple =  3;
        
        launchView.lastDotsIsHidden = YES;//最后一个页面时是否隐藏PageControl 默认为NO
        launchView.pageDotImage = [UIImage imageNamed:@"Ktv_ic_share_qq"];
        launchView.currentPageDotImage = [UIImage imageNamed:@"Ktv_ic_share_weixin"];
        launchView.loadFinishBlock = ^(MSLaunchView * _Nonnull launchView) {
            NSLog(@"广告加载完成了");
        };
        
#pragma mark ---->>自定义立即体验按钮
        [launchView guideBtnCustom:^UIButton * _Nonnull{
            UIButton *btn = [UIButton buttonWithType:UIButtonTypeCustom];
            btn.frame = CGRectMake(60, 60, 130, 60);
            [btn setBackgroundColor:[UIColor redColor]];
            [btn setTitle:@"立即体验" forState:UIControlStateNormal];
            [btn addTarget:self action:@selector(hidde) forControlEvents:UIControlEventTouchUpInside];
            return btn;
        }];
        
#pragma mark ---->>自定义跳过按钮

//        [launchView skipBtnCustom:^UIButton * _Nonnull{
//            UIButton *btn = [UIButton buttonWithType:UIButtonTypeCustom];
//            btn.frame = CGRectMake(60, 200, 120, 120);
//            [btn setBackgroundColor:[UIColor blueColor]];
//            [btn setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
//            [btn setTitle:@"跳过按钮" forState:UIControlStateNormal];
//            [btn addTarget:self action:@selector(hidde) forControlEvents:UIControlEventTouchUpInside];
//            return btn;
//        }];

        _launchView = launchView;
    }

    return YES;
}

-(void)hidde{
    [_launchView hideGuidView];
}

-(void)launchViewLoadFinish:(MSLaunchView *)launchView{
    NSLog(@"代理方法进入======广告加载完成了");
}

- (void)applicationWillResignActive:(UIApplication *)application {
    // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
    // Use this method to pause ongoing tasks, disable timers, and invalidate graphics rendering callbacks. Games should use this method to pause the game.
}


- (void)applicationDidEnterBackground:(UIApplication *)application {
    // Use this method to release shared resources, save user data, invalidate timers, and store enough application state information to restore your application to its current state in case it is terminated later.
    // If your application supports background execution, this method is called instead of applicationWillTerminate: when the user quits.
}


- (void)applicationWillEnterForeground:(UIApplication *)application {
    // Called as part of the transition from the background to the active state; here you can undo many of the changes made on entering the background.
}


- (void)applicationDidBecomeActive:(UIApplication *)application {
    // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
}


- (void)applicationWillTerminate:(UIApplication *)application {
    // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
}


@end
